src/ol/renderer/webgl/shaders.js

```diff
@@ -0,0 +1,213 @@
+/**
+ * @module ol/renderer/webgl/shaders
+ */
+import {asArray} from '../../color.js';
+
+/**
+ * Attribute names used in the default shaders.
+ * @enum {string}
+ */
+export const DefaultAttributes = {
```

Comment:
It would be nice to not introduce new string enums. Using a union type like
```js
/** @typedef {'color'|'opacity'|'width'} DefaultAttributes */
```
would also make the example look nicer, if the shader`s attribute were typed like
```js
/** @typedef {Object<DefaultAttributes, (import("ol/Feature").default, Object<string, *>) => number} ShaderAttributes */
```

Reply:
I agree.  I've pushed another commit that makes this change.